alpha/apps/kaltura/lib/storage/urlTokenizers/kWowzaSecureTokenUrlTokenizer.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+class kWowzaSecureTokenUrlTokenizer extends kUrlTokenizer
+{
+	
+	/**
+	 * @var string
+	 */
+	protected $paramPrefix;
+	
+	/**
+	 * @var bool
+	 */
+	protected $shouldIncludeClientIp;
+	
+	/**
+	 * @var string
+	 */
+	protected $hashAlgorithm;
+
+	
+	public function tokenizeSingleUrl($url, $urlPrefix = null)
+	{
+		$tokenPrefix = $this->getParamPrefix();
+		$expiryTimeFrame = $this->getWindow();
+		
+		$params = array();
+		
+		$params["starttime"] = time();
+		$params["endtime"] = time() + $expiryTimeFrame;
+		
+		$prefixedParams = $this->prefixParams($tokenPrefix,$params);
+		
+		return $url."?".http_build_query($prefixedParams,'','&').'&'.$tokenPrefix.'hash='.$this->getHash($prefixedParams,$url);
+		
+	}
+	
+	/**
+     * Get prefixed parameters
+     * 
+     * @return array
+     */
+	protected function prefixParams($prefix,$params){
+		
+		if(!empty($prefix))
+        {
+            foreach($params as $key => $param)
+            {
+                if(strpos($key, $prefix) === false)
+                {
+                    $params[$prefix . $key] = $param;
+                    unset($params[$key]);
+                }
+            }
+        }
+        
+        return $params;
+	}
+		
+	/**
+     * Get hash token
+     * 
+     * @return string
+     */
+    public function getHash($params,$url){
+	
+		$tokenKey = $this->getKey();
+		$shouldIncludeClientIp = $this->getShouldIncludeClientIp();
+				
+		$params[$tokenKey] = "";
+		
+		if($shouldIncludeClientIp == true){
+			$params[self::getRemoteAddress()] = "";
+		}
+        
+        ksort($params);
+		
+        $query = '';
+        foreach($params as $k => $v)
+        {
+            $query .= '&' . $k;
+            if(isset($v) && !empty($v))
+            {
+                $query .= '=' . $v;
+            }
+        }
+		
+        $query = trim($query, '&');
+		
+		$urlInfo = parse_url($url);
+		
+        $path = ltrim($urlInfo['path'], '/');
+        $pathItems = explode('/', $path);
+
+        $path = "";
+        foreach ($pathItems as $k => $pathItem) {
+            if(1 === preg_match('/(^Manifest|\.m3u8|\.f4m|\.mpd)/',$pathItem)){
+                break;
+            }
+            $path .= $pathItem;
+            if(count($pathItems)-1 != $k) {
+                $path .= '/';
+            }
+        }
+        if(strrpos($path, '/') === strlen($path)-1) {
```

Comment:
Can be replaced with rtrim, or better - remove the ltrim before the loop and just do trim here to remove both